dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat/fat/src/com/ibm/websphere/microprofile/faulttolerance_fat/tests/stateless/BarrierFactory.java

```diff
@@ -0,0 +1,105 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package com.ibm.websphere.microprofile.faulttolerance_fat.tests.stateless;
+
+import static org.hamcrest.Matchers.empty;
+import static org.junit.Assert.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * <pre>
+ * <code>
+ * try (BarrierFactory bf = new BarrierFactory()) {
+ *     Barrier b = bf.create();
+ *     Future f = runAsync(bean.run(b));
+ *     assertNotComplete(f);
+ *     b.complete();
+ *     assertCompletes(f);
+ * }
+ * </code>
+ * </pre>
+ *
+ */
+public class BarrierFactory implements AutoCloseable {
+
+    /** Timeout for awaiting barriers, should be longer than any test requires */
+    private static final int TIMEOUT = 10000;
+    /** Timeout for waiting for await() to be called on a barrier */
+    private static final int COMPLETION_TIMEOUT = 1000;
```

Comment:
Need to check these timeout values vs. what we can depend on in our build systems.